test_util/test_upgrade_vpc.py

```diff
@@ -360,6 +379,104 @@ def test_app_dns_survive_upgrade():
             )
             self.log_test("test_upgrade_vpc.test_app_dns_survive_upgrade", test_app_dns_survive_upgrade)
 
+    def mesos_metrics_snapshot(self, cluster, host):
+        raise NotImplementedError()
+
+    def upgrade_host(self, tunnel, role, bootstrap_url, upgrade_script_path):
+        # Download the upgrade script for the new DC/OS.
+        tunnel.remote_cmd(curl_cmd + ['--remote-name', upgrade_script_path])
+
+        # Upgrade to the new DC/OS.
+        tunnel.remote_cmd(['sudo', 'bash', 'dcos_node_upgrade.sh'])
+
+    @retry(
+        wait_fixed=(1000 * 5),
+        stop_max_delay=(1000 * 60 * 5),
+        # Retry on SSH command error or metric not equal to expected value.
+        retry_on_exception=(lambda exc: isinstance(exc, CalledProcessError)),
+        retry_on_result=(lambda result: not result),
+    )
+    def wait_for_mesos_metric(self, cluster, host, key, value):
+        """Return True when host's Mesos metric key is equal to value."""
+        return self.mesos_metrics_snapshot(cluster, host).get(key) == value
+
+    def upgrade_dcos(self, cluster, installer_url, version, add_config_path=None):
```

Comment:
This function may be useful outside the context of an upgrade test. It should stay in the `cluster` lib.